frontend/src/components/navTabs.tsx

```diff
@@ -25,33 +26,73 @@ const NavTabs = (props: Props) => {
   const navigate = useNavigate();
   const activeTab = tabs.find((tab) => location.pathname.includes(tab.path));
   const activeTabPath = activeTab?.path || "";
+
+  // State to manage the visibility of the mobile menu
+  const [mobileMenuVisible, setMobileMenuVisible] = useState(false);
+
+  // Toggle the mobile menu
+  const toggleMobileMenu = () => {
+    setMobileMenuVisible(!mobileMenuVisible);
+  };
+
   return (
-    <>
-      {tabs.map((tab) => (
-        <NavLink
-          key={`${tab.label}-${tab.path}`}
-          active={
-            tab.isActive !== undefined
-              ? tab.isActive(tab.path)
-              : activeTabPath === tab.path
-          }
-          label={tab.label}
-          onClick={() => {
-            onChange?.(tab.path);
-            tab.onClick?.();
-            tab.path && navigate(tab.path);
-          }}
-          icon={tab?.Icon}
-          disabled={props.disabled || tab.disabled}
-        >
-          {tab.nestedTabs && (
-            <Box mb={2}>
-              <NavTabs onChange={onChange} tabs={tab.nestedTabs} />
-            </Box>
-          )}
-        </NavLink>
-      ))}
-    </>
+    <nav>
+      <div className="desktop-nav">
+        {tabs.map((tab) => (
+          <NavLink
+            key={`${tab.label}-${tab.path}`}
+            active={
+              tab.isActive !== undefined
+                ? tab.isActive(tab.path)
+                : activeTabPath === tab.path
+            }
+            label={tab.label}
+            onClick={() => {
+              onChange?.(tab.path);
+              tab.onClick?.();
+              tab.path && navigate(tab.path);
+            }}
+            icon={tab?.Icon}
+            disabled={props.disabled || tab.disabled}
+          >
+            {tab.nestedTabs && (
+              <Box mb={2}>
+                <NavTabs onChange={onChange} tabs={tab.nestedTabs} />
+              </Box>
+            )}
+          </NavLink>
+        ))}
+      </div>
+      <div className="mobile-nav">
+        <button className="hamburger-menu" onClick={toggleMobileMenu}>
+          ☰
```

Comment:
For icons we are using "@tabler/icons-react" library, You can search for the icon here easily - https://tabler-icons-react.vercel.app/.

Here we can use `import { IconMenu2 } from "@tabler/icons-react";`